src/backend/resourcemanager/communication/rmcomm_QD2RM.c

```diff
@@ -2145,7 +2145,7 @@ int findFile(const char *filename)
 int createFile(const char *filename)
 {
 	int fd = 0;
-	fd = open(filename, O_RDWR|O_CREAT);
+	fd = open(filename, O_RDWR|O_CREAT, S_IRUSR | S_IWUSR);
```

Comment:
The S_IRUSR and S_IWUSR flags indicate that the file should be created with the u+rw permission bits set, presumably default permissions were different between Ubuntu and CentOS and switching to the three parameter open() call is more explicit in the behavior.